### DISCUSSION TITLE
把一个 websocket id 移交到另一个服务

### DISCUSSION THREAD

**Discussion by korialuo** [cloudwu/skynet#1604] [General]
您好，云风老师。skynet 现有的 websocket 库，没有办法把一个已经建立好的连接( ws_id )，转交给另一个服务使用。按照我之前的 skynet.socket 的用法，我在 gate 服务上进行了密钥交换，然后把 socket_fd 移交到另外一个 agent 服务上使用。但是现在项目换成了 websocket， 在 gate 上协商好密钥过后，就不能再把 ws_id 交到 agent 服务上了。只能把发到客户端的消息转发到 gate 上，由 gate 负责 websocket 的发送。我不知道这样会不会在 gate 上形成瓶颈，另外云风老师觉得有没必要给 websocket 扩展一个类似 socket.abandom() 这样的方法？

**Comment by cloudwu** [comment]
你可以做一个或多个同样的服务专门处理 websocket 连接。

**Comment by hanxi** [comment]
https://github.com/hanxi/skynet-demo/blob/master/service/ws_gate.lua

担心 gate 成为瓶颈的话可以看看我以前写的多 gate 的例子，你可以参考一下。
